.evergreen/config.yml

```diff
@@ -16,33 +43,146 @@ functions:
         add_to_path:
           - .evergreen
         script: |
+          ${PREPARE_SHELL}
           ${script}
 
+  "prepare resources":
+    - command: subprocess.exec
+      params:
+        working_dir: src
+        binary: bash
+        add_to_path:
+          - .evergreen
+        args:
+          - fetch-drivers-tools.sh
+        include_expansions_in_env:
+          - DRIVERS_TOOLS
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
+
+  "bootstrap mongo-orchestration":
+    - command: subprocess.exec
+      params:
+        binary: sh
+        args:
+          - ${DRIVERS_TOOLS}/.evergreen/run-orchestration.sh
+        include_expansions_in_env:
+          - AUTH
+          - SSL
+          - TOPOLOGY
+          - REQUIRE_API_VERSION
+          - MONGODB_VERSION
+          - ORCHESTRATION_FILE
+          - MONGODB_BINARIES
+    - command: expansions.update
+      params:
+        file: mo-expansion.yml
+
+  "stop mongo orchestration":
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          cd ${MONGO_ORCHESTRATION_HOME}
+          # source the mongo-orchestration virtualenv if it exists
+          if [ -f venv/bin/activate ]; then
+            . venv/bin/activate
+          elif [ -f venv/Scripts/activate ]; then
+            . venv/Scripts/activate
+          fi
+          mongo-orchestration stop || true
+
+  "cleanup":
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          rm -rf $DRIVERS_TOOLS || true
+
+# Functions to run before all tasks (except those in task groups).
 pre:
-  - func: fetch source
+  - func: "fetch source"
+  - func: "create expansions"
+  - func: "prepare resources"
 
+# Functions to run after all tasks (except those in task groups).
 post:
+  - func: "stop mongo orchestration"
```

Comment:
I did think about this idea. Moving this step to task has the danger that if preceding task steps failed, it might be skipped. Adding it to global post won't hurt (as Rust driver team does) for it seems to return if mongoDB has not been started.
I'll investigate further.

Reply:
Per https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Task-Runtime-Behavior:
<img width="971" alt="Screenshot 2024-11-08 at 10 21 06 AM" src="https://github.com/user-attachments/assets/a62f33be-48f0-4827-9c95-bb27418e33fc">

it seems we have to put such resoruce cleanup stuff into `post`, not as the last step of Main task commands.
Even in our driver setup, we put it into `clearup` which will be run in `post`.

Reply:
ok, let's leave it as is then